From bdfca8691141ac04af1c685a7032e23706def6cc Mon Sep 17 00:00:00 2001
From: Anders Berg <anders.berg@intel.com>
Date: Wed, 17 Dec 2014 11:46:47 +0100
Subject: [PATCH 006/131] arm64: dts: Corrected SPI definitions for AXM56xx

The Axxia SSP needs to force the peripheral ID in the device tree to
make the driver use the chip select extensions available in this PL022
variant.

Signed-off-by: Anders Berg <anders.berg@intel.com>
[Original patch is from linux-yocto-4.1 axxia/base branch.]
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 arch/arm64/boot/dts/intel/axm56xx.dtsi | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/intel/axm56xx.dtsi b/arch/arm64/boot/dts/intel/axm56xx.dtsi
index 9e10079..ea040ae 100644
--- a/arch/arm64/boot/dts/intel/axm56xx.dtsi
+++ b/arch/arm64/boot/dts/intel/axm56xx.dtsi
@@ -46,7 +46,7 @@
 
 		clks: dummy-clock {
 			compatible = "fixed-clock";
-			#clock-cells = <0>;
+			#clock-cells = <1>;
 			clock-frequency = <100000000>;
 		};
 	};
@@ -370,6 +370,7 @@
 
 			spi0: ssp@8080100000 {
 				compatible = "arm,pl022", "arm,primecell";
+				arm,primecell-periphid = <0x000b6022>;
 				#address-cells = <1>;
 				#size-cells = <0>;
 				reg = <0x80 0x80100000 0x1000>;
-- 
2.8.1

